test/org/sagebionetworks/bridge/services/TaskServiceMockTest.java

```diff
@@ -149,14 +166,50 @@ public void rejectsListOfTasksWithTaskThatLacksGUID() {
         service.updateTasks("AAA", tasks);
     }
     
+    @SuppressWarnings("unchecked")
     @Test
     public void updateTasksWorks() {
         ScheduleContext context = createScheduleContext(endsOn);
         List<Task> tasks = TestUtils.runSchedulerForTasks(user, context);
         
+        // 4 tasks, finish two, these should publish events, the third with a startedOn 
+        // timestamp will be saved, so 3 tasks sent to the DAO
+        assertEquals(4, tasks.size());
+        tasks.get(0).setStartedOn(DateTime.now().getMillis());
+        tasks.get(1).setFinishedOn(DateTime.now().getMillis());
+        tasks.get(2).setFinishedOn(DateTime.now().getMillis());
+        
+        ArgumentCaptor<List> argument = ArgumentCaptor.forClass(List.class);
+        
+        service.updateTasks("BBB", tasks);
+        
+        verify(taskDao).updateTasks(anyString(), argument.capture());
+        verify(taskDao, times(3)).getTask(anyString(), anyString());
+        verify(taskEventService, times(2)).publishTaskFinishedEvent(any());
```

Comment:
Do we want to check that the tasks passed into publishTaskFinishedEvent() are the ones we expect?